src/compiler/emitter.ts

```diff
@@ -604,11 +604,13 @@ namespace ts {
             pipelinePhase(hint, node);
         }
 
-        function setSourceFile(sourceFile: SourceFile) {
+        function setSourceFile(sourceFile: SourceFile | undefined) {
             currentSourceFile = sourceFile;
             currentLineMap = undefined;
             detachedCommentsInfo = undefined;
-            setSourceMapSource(sourceFile);
+            if (sourceFile) {
+                setSourceMapSource(sourceFile);

```

Comment:
Don't you need to reset it in sourceMap as well (to avoid accidental wrong sourceFile pick)?

Reply:
`setSourceMapSource` very much does not handle `undefined` input. `sourceMapSource` is only used when emitting sourcemaps which is only used when descending into an actual file (emitting a node with a sourcemap pipeline phase, actually - versus printing comments at the top level of a bundle, where we set the current source file so we can get the text, and need to unset it before we call a function which uses it set-ed-ness to determine if it needs to print). It doesn't need to be unset partially because it expects to be defined in all places where it is used (unlike `currentSourceFile` which is actually checked).